doc/src/arch/reference.rst

```diff
@@ -2036,3 +2036,42 @@ The full format is documented below.
     This specifies that the 'from' set is the union of L4 switchpoints 0, 1, 2 and 3; and L16 switchpoints 0, 4, 8 and 12.
     The 'to' set is all L4 switchpoint 0's.
     Note that since different switchpoints are selected from different segment types it is not possible to specify this without using ``<from>`` sub-tags.
+
+.. _arch_metadata:
+
+Architecture metadata
+---------------------
+
+To enable tagging of architecture structures with metadata, the ``<metadata>``
+tag can be inserted under the following XML tags:
+
+* ``<pb_type>``
+* Any tag under ``<interconnect>`` (``<direct>``, ``<mux>``, etc).
+* ``<mode>``
+* Any grid location type (``<perimeter>``, ``<fill>``, ``<corners>``, ``<single>``, ``<col>``, ``<row>``, ``<region>``)
+
+.. arch:tag:: <metadata>
+
+Specifies the root of a metadata block.  Can have 0 or more ``<meta>`` Children.
+
+.. arch:tag:: <meta name="string" >
+
+    :req_param name: Key name of this metadata value.
+
+Specifies a value within a metadata block.  The name is a key
+for looking up the value contained within the ``<meta>`` tag.  Keys can be
+repeated, and will be stored in a vector in order of occurance.
+
+The value of the ``<meta>`` is the text in the block. Both the ``name`` and
+``<meta>`` value will be stored as a string.  XML children are not supported
+in the ``<meta>`` tag.
+
+Example of a metadata block with 2 keys:
+
+    .. code-block:: xml
+
+      <metadata>
```

Comment:
Looking at the code, it looks like using the same key multiple times is OK, and you'll just append all the values into a vector for that key.  I think somewhere in the documentation it should mention what is and isn't allowed (e.g. repeated keys are allowed; their values will be appended to be stored with that key).  Or if you don't allow repeated keys (breaks something else) say that.

Reply:
There is already a comment to this affect?

> Keys can be repeated, and will be stored in a vector in order of occurrence.